tests/test_frame.py

```diff
@@ -2467,3 +2467,19 @@ def test_astype() -> None:
 
     states = pd.DataFrame({"population": population, "area": area})
     check(assert_type(states.astype(object), pd.DataFrame), pd.DataFrame, object)
+
+
+def test_xs_frame_new() -> None:
+    d = {
+        "num_legs": [4, 4, 2, 2],
+        "num_wings": [0, 0, 2, 2],
+        "class": ["mammal", "mammal", "mammal", "bird"],
+        "animal": ["cat", "dog", "bat", "penguin"],
+        "locomotion": ["walks", "walks", "flies", "walks"],
+    }
+    df = pd.DataFrame(data=d)
+    df = df.set_index(["class", "animal", "locomotion"])
+    s1 = df.xs("mammal", axis=0)
+    s2 = df.xs("num_wings", axis=1)
+    check(assert_type(s1, Union[pd.Series, pd.DataFrame]), pd.DataFrame)
+    check(assert_type(s2, Union[pd.Series, pd.DataFrame]), pd.DataFrame)
```

Comment:
Should be `check(assert_type(s2, Union[pd.Series, pd.DataFrame]), pd.Series)`

Reply:
Done with it but i could not understand why pd.Series instead of DataFrame, can you please explain

Reply:
Just run the code in python:
```python
>>> import pandas as pd
>>> d = {
...         "num_legs": [4, 4, 2, 2],
...         "num_wings": [0, 0, 2, 2],
...         "class": ["mammal", "mammal", "mammal", "bird"],
...         "animal": ["cat", "dog", "bat", "penguin"],
...         "locomotion": ["walks", "walks", "flies", "walks"],
...     }
>>> df = pd.DataFrame(data=d)
>>> df = df.set_index(["class", "animal", "locomotion"])
>>> s2 = df.xs("num_wings", axis=1)
>>> type(s2)
<class 'pandas.core.series.Series'>
>>> s2
class   animal   locomotion
mammal  cat      walks         0
        dog      walks         0
        bat      flies         2
bird    penguin  walks         2
Name: num_wings, dtype: int64
```
It's just how `xs` works.  In this particular case, you are getting a particular column of `df`.  But `axis=1` can return a `DataFrame`:
```python
>>> df.T.xs("mammal", axis=1)
animal       cat   dog   bat
locomotion walks walks flies
num_legs       4     4     2
num_wings      0     0     2
```

That's why the type is a `Union`, but the result at runtime is a `Series`

Reply:
Thanks sir